src/Orleans/Streams/PersistentStreams/IStreamQueueBalancer.cs

```diff
@@ -8,27 +8,32 @@ namespace Orleans.Streams
     /// The stream queue balancer is responsible for load balancing queues across all other related queue balancers.  It
     /// notifies any listeners (<code>IStreamQueueBalanceListener</code>) of changes to the distribution of queues.
     /// </summary>
-    internal interface IStreamQueueBalancer
+    public interface IStreamQueueBalancer
     {
         /// <summary>
+        /// Async Initialize logic
+        /// </summary>
+        /// <returns></returns>
+        Task Initialize();
```

Comment:
add hock for async initialize logic, such as load queue balancer from an async storage service, or connect async lease manager for initialize logic. 

The same reason for other methods to be async